pan-domain-node/src/api.ts

```diff
@@ -0,0 +1,31 @@
+export { PanDomainAuthentication } from './panda';
+
+export enum AuthenticationStatus {
+    INVALID_COOKIE = 'Invalid Cookie',
+    EXPIRED = 'Expired',
+    NOT_AUTHORISED = 'Not Authorised',
+    AUTHORISED = 'Authorised'
+}
+
+export interface User {
+    firstName: string,
+    lastName: string,
+    email: string,
+    avatarUrl?: string,
+    authenticatingSystem: string,
+    authenticatedIn: string[],
+    expires: number,
+    multifactor: boolean
+}
+
+export interface AuthenticationResult {
+    status: AuthenticationStatus,
+    user?: User 
+}
+
+export type ValidateUserFn = (user: User) => boolean;
+
+export function guardianValidation(user: User): boolean {
```

Comment:
would `isGuardianUser` be more intention revealing?

Reply:
Possibly. I named it the same as the [function in the Scala library](https://github.com/guardian/pan-domain-authentication/blob/master/pan-domain-auth-verification/src/main/scala/com/gu/pandomainauth/PanDomain.scala#L52) just for API consistency

Reply:
Don't like it, but the reasoning makes sense.